src/logic/calculate.js

```diff
@@ -1,21 +1,47 @@
 import operate from './operate';
 
 const calculate = (btnName, dataObj) => {
-  let { total } = dataObj;
-  const { next, operation } = dataObj;
+  const { total, next, operation } = dataObj;
```

Comment:
Good job making a correction from the previous review.

It seems redundant to hold `total, next, operation` in another variable.

you can just destructure use `let` then use them in your code, right? 

Example: 

`let { total, next, operation } = dataObj;`

```
  if (btnName === 'AC') {
    total = null;
```